File: 백준/Bronze/21756. 지우개/지우개.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin");
// let input = fs.readFileSync("./input.txt");
input = input.toString().trim();

const N = parseInt(input[0]);
let arr = new Array(N + 1).fill(0).map((x, i) => (x = i));

if (arr.length <= 2) {
  box = arr;
}
while (arr.length > 2) {
  box = [];
  arr.forEach((x, i) => {
    if (i % 2 == 0) {
      box.push(x);
    }
  });
  arr = box;
}

console.log(box[1]);

File: 백준/Bronze/21756. 지우개/지우개.py
import sys
# sys.stdin = open("input.txt", 'r')
input = sys.stdin.readline

N = int(input())
arr = [i for i in range(N+1)]

if len(arr) <= 2:
    box = arr
while len(arr) > 2:
    box = []
    for i in range(len(arr)):
        if i % 2 == 0:
            box.append(arr[i])
    arr = box

print(box[1])